ouroboros-consensus/test-consensus/Test/Consensus/Protocol/PBFT.hs

```diff
@@ -24,220 +25,312 @@ import           Ouroboros.Network.Block (SlotNo (..))
 import           Ouroboros.Network.Point (WithOrigin (..))
 
 import           Ouroboros.Consensus.Protocol.Abstract
-import           Ouroboros.Consensus.Protocol.PBFT.ChainState (EbbMap (..), PBftChainState)
+import           Ouroboros.Consensus.Protocol.PBFT.ChainState (EbbMap (..),
+                     PBftChainState)
 import qualified Ouroboros.Consensus.Protocol.PBFT.ChainState as CS
 import           Ouroboros.Consensus.Protocol.PBFT.Crypto
-import           Ouroboros.Consensus.Util (dropLast, repeatedly, takeLast)
+import           Ouroboros.Consensus.Util (repeatedly)
 
+import           Test.Util.Split (spanLeft, splitAtJust)
 
 {-------------------------------------------------------------------------------
   Top-level tests
 -------------------------------------------------------------------------------}
 
 tests :: TestTree
-tests = testGroup "PBftChainState" [
-      testProperty "validGenerator"                   prop_validGenerator
-    , testProperty "appendPreservesInvariant"         prop_appendPreservesInvariant
-    , testProperty "rewindPreservesInvariant"         prop_rewindPreservesInvariant
-    , testProperty "rewindReappendId"                 prop_rewindReappendId
-    , testProperty "appendOldStatePreservesInvariant" prop_appendOldStatePreservesInvariant
-    , testProperty "appendOldStateRestoresPreWindow"  prop_appendOldStateRestoresPreWindow
+tests = testGroup "PBftChainState" $
+    -- these tests are cheap, so inflate the count
+    map (adjustOption $ \(QuickCheckTests n) -> QuickCheckTests $ 10 * n) $
```

Comment:
Fixed.

Reply:
This could inspire a QuickCheck feature request: a couple command-line parameters that

  * specify a total desired run-time

  * a minimum ratio from longest test to shortest test.

QC would inflate test counts for the quickest tests until the ratio was met. Then it would evenly increase all test counts (~batched round-robin?) until the desired total time expired.

Reply:
I would :+1: that feature request :smile: